cEP-0017.md

```diff
@@ -0,0 +1,366 @@
+# cib Design and Improvements
+
+| Metadata |                                                     |
+| -------- | --------------------------------------------------- |
+| cEP      | 17                                                  |
+| Version  | 1.0                                                 |
+| Title    | cib Design and Improvements                         |
+| Authors  | Saksham Bansal <mailto:sakshambansal1999@yahoo.com> |
+| Status   | Implementation due                                  |
+| Type     | Process                                             |
+
+## Abstract
+
+This cEP describes changes to the design of `cib` which will allow it to install
+individual bears and their dependencies. It suggests using conda packages for bears
+that have different installation instructions for each distribution and enhancing
+of the existing `Requirement` classes.
+
+The cEP discusses Ansible which is a simple IT automation software as a means
+to install bears on different distributions by creating a playbook with the required
+installation instructions for all bears.
+
+## Introduction
+
+The current design of the cib tool allows it to perform basic operations such as
+running pip install on bears and then installing its dependencies. However, some
+bears are not even getting installed on certain distributions and cib is running into
+errors. Moreover, `cib` is not able to handle special cases such as of bears
+that have different installation instructions for each distribution
+i.e. `DartLintBear` and bears that have bear dependencies such as the
+`ClangCloneDetectionBear`. Additionally, `cib` needs to be more user friendly
+with a nice interface and should be able to provide verbose output,
+ask user for permission before proceeding.
+
+Another alternative to `cib` that can be used for installing bears and their dependencies
+is Ansible. Ansible allows to orchestrate steps of any process
+for different distributions by creating `playbooks` in YAML.
```

Comment:
playbooks isnt a code literal -- it is a English term created by Ansible, so should be in double quotes `"playbooks"`